File: lc_py/codeforces/math/mexpartition.py
import sys
def solve(a):
    cnt = [0] * 102
    for x in a:
        cnt[x] += 1
    mex = 0
    for i in range(102):
        if cnt[i] == 0:
            mex = i 
            break
    return mex

if __name__ == "__main__":
    input = sys.stdin.read
    data = input().split()
    t = int(data[0])
    idx = 1
    for _ in range(t):
        n = int(data[idx])
        idx += 1
        a = [int(x) for x in data[idx:idx + n]]
        idx += n
        print(solve(a))
        

File: lc_go/codeforces/ds/math/mexPartition.go
package main

import (
	"fmt"
)

var (
	t, n int
	a    []int
)

func solve(a []int) int {
	var mex int = 0
	// n := len(a)
	cnt := make([]int, 102)
	for _, x := range a {
		cnt[x]++
	}
	for i := 0; i <= 101; i++ {
		if cnt[i] == 0 {
			mex = i
			break
		}
	}
	return mex
}
func main() {
	fmt.Scan(&t)
	for ; t > 0; t-- {
		fmt.Scan(&n)
		a = make([]int, n)
		for i := 0; i < n; i++ {
			fmt.Scan(&a[i])
		}
		fmt.Println(solve(a))
	}

}
